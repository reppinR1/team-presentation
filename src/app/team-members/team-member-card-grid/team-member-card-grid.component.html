<ng-container *ngIf="memberCards; else loading">
  <section class="fluid-grid">
    <app-team-member-card
      [memberCard]="memberCard.value"
      *ngFor="let memberCard of memberCards | keyvalue"
    >
    </app-team-member-card>
  </section>
</ng-container>
<ng-template #loading> <section>Loading...</section></ng-template>
