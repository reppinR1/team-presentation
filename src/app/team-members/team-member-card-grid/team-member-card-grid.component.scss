:host {
  display: block;
  width: 100%;
}

.fluid-grid {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(240px, 1fr));
  grid-gap: 8px;
  justify-items: center;
}
