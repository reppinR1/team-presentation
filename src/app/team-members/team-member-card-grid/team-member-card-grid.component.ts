import { Component, Input } from '@angular/core';
import { MemberCards } from '../model';

@Component({
  selector: 'app-team-member-card-grid',
  templateUrl: './team-member-card-grid.component.html',
  styleUrls: ['./team-member-card-grid.component.scss'],
})
export class TeamMemberCardGridComponent {
  @Input() memberCards: MemberCards | null;
}
