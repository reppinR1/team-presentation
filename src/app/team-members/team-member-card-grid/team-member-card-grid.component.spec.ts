import { TeamMemberCardGridComponent } from './team-member-card-grid.component';
import { createComponentFactory, Spectator } from '@ngneat/spectator';

describe('TeamMemberCardGridComponent', () => {
  let spectator: Spectator<TeamMemberCardGridComponent>;
  const createComponent = createComponentFactory({
    component: TeamMemberCardGridComponent,
    shallow: true,
  });

  beforeEach(() => {
    spectator = createComponent({
      props: { memberCards: [] as any },
    });
  });

  it('should should match snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });
});
