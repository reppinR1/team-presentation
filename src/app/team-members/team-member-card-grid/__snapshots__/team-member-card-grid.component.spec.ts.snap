// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMemberCardGridComponent should should match snapshot 1`] = `
<app-team-member-card-grid
  memberCards={[Function Array]}
>
  <section
    class="fluid-grid"
  >
    
  </section>
</app-team-member-card-grid>
`;
